src/frontend/screens/ObservationEdit/useMostAccurateLocationForObservation.ts

```diff
@@ -20,13 +20,21 @@ export function useMostAccurateLocationForObservation() {
   const locationServicesTurnedOff =
     providerStatus && !providerStatus.locationServicesEnabled;
 
-  if (locationServicesTurnedOff && !value?.metadata.position) {
+  const isLocationManuallySet = !!value?.metadata.manualLocation;
+
+  // If location services are turned off (and the observation location is not manually set),
+  // we want to immediately update the draft so that this hook does not return a stale position
+  if (
+    locationServicesTurnedOff &&
+    value?.metadata.position &&
+    !isLocationManuallySet
+  ) {
     updateObservationPosition({position: undefined, manualLocation: false});
   }
 
   useFocusEffect(
     useCallback(() => {
-      if (!permissions || !permissions.granted) return;
+      if (!permissions || !permissions.granted || isLocationManuallySet) return;
```

Comment:
We also should check if the `watchPositionAsync` is running, and if it is turn if off. In most cases, the subscription is turned on, then location is manually set.

Reply:
not quite following - isn't this handled in the cleanup of this effect?

```ts
      return () => {
        ignore = true;
        locationSubscriptionProm.then(sub => sub.remove());
      }
```

Reply:
Oh right, i always forget that the cleanup effect gets called on every change not just on dismount.

Reply:
yeah in this case i think the behavior is slightly different since it's `useFocusEffect` but i think it's okay to let it handle the subscription removal - can follow up if there turns out to be a major issue